src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -573,12 +573,76 @@ void ResumptionDataProcessor::AddCommands(
       application, application_manager_));
 }
 
+utils::Optional<ResumptionRequest> FindCommandRequest(
+    uint32_t command_id, std::vector<ResumptionRequest>& requests) {
+  using namespace utils;
+  auto request_it = std::find_if(
+      requests.begin(),
+      requests.end(),
+      [command_id](const ResumptionRequest& request) {
+        if (request.message[strings::msg_params].keyExists(strings::cmd_id) &&
+            (request.message[strings::msg_params].keyExists(
+                 strings::menu_params) ||
+             (request.message[strings::msg_params][strings::type] ==
+              hmi_apis::Common_VRCommandType::Command))) {
+          uint32_t cmd_id =
+              request.message[strings::msg_params][strings::cmd_id].asUInt();
+          return cmd_id == command_id;
+        }
+        return false;
+      });
+  if (requests.end() != request_it) {
+    return Optional<ResumptionRequest>(*request_it);
+  }
+  return Optional<ResumptionRequest>::OptionalEmpty::EMPTY;
+}
+
 void ResumptionDataProcessor::DeleteCommands(ApplicationSharedPtr application) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const uint32_t app_id = application->app_id();
+  resumption_status_lock_.AcquireForReading();
+  std::vector<ResumptionRequest> failed_requests;
+  std::vector<ResumptionRequest> missed_requests;
+  if (resumption_status_.find(app_id) != resumption_status_.end()) {
+    failed_requests = resumption_status_[app_id].error_requests;
+    missed_requests = resumption_status_[app_id].list_of_sent_requests;
+  }
+  resumption_status_lock_.Release();
+
+  failed_requests.insert(
+      failed_requests.end(), missed_requests.begin(), missed_requests.end());
+
   app_mngr::CommandsMap cmap = application->commands_map().GetData();
 
   for (auto cmd : cmap) {
-    MessageHelper::SendDeleteCommandRequest(
-        cmd.second, application, application_manager_);
+    auto failed_command = FindCommandRequest(cmd.first, failed_requests);
+    bool VR_failed = false;
+    bool UI_failed = false;
+    if (failed_command) {
+      (*failed_command)
+              .message[strings::msg_params]
```

Comment:
create lambda 
auto is_vr_command = [](SmartObject) {
}
and use it instead of boolean flags

Reply:
Please look at e75db783045c452a32c33b86b646ff3cd2006d5b